### ISSUE TITLE
Invalid warning building Analytics with Swift Package Manager for macOS

### ISSUE THREAD

**Issue by Overcot** [firebase/firebase-ios-sdk#9029]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13.1
  * Firebase SDK version: 8.9.1
  * Installation method: `Swift Package Manager` (select one)
  * Firebase Component: FirebaseFunctionsWrap
  * Target platform(s): `All`

### [REQUIRED] Step 2: Describe the problem
When I compile Firebase modules I receive several warnings regarding FirebaseFunctionsWrap

#### Steps to reproduce:

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.

I have an spm package that links firebase like this
```swift
  dependencies: [
    // Dependencies declare other packages that this package depends on.
    .package(name: "Firebase", url: "https://github.com/firebase/firebase-ios-sdk.git", .upToNextMajor(from: "8.9.1")),
  ],
  targets: [
    .target(
      name: "SAFAnalytics",
      dependencies: [
        .product(name: "FirebaseAnalytics", package: "Firebase"),
        .product(name: "FirebaseCrashlytics", package: "Firebase"),
        .product(name: "FirebaseDynamicLinks", package: "Firebase"),
        .product(name: "FirebaseInstallations", package: "Firebase"),
        .product(name: "FirebaseMessaging", package: "Firebase"),
        .product(name: "FirebaseRemoteConfig", package: "Firebase"),
      ],
    ),
    .testTarget(name: "SAFAnalyticsTests", dependencies: ["SAFAnalytics"]),
  ]
```
When I compile it inside AppCode or Xcode using command

```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift build --skip-update --build-tests --build-path /Users/overcot/Work/kinbery-ios/SAF/SAFAnalytics/.build
```
it produces a lot of warnings

/Users/overcot/Work/kinbery-ios/SAF/SAFAnalytics/.build/arm64-apple-macosx/debug/FirebaseFunctionsTarget.build/module.modulemap:2:14: warning: umbrella directory '/Users/overcot/Work/kinbery-ios/SAF/SAFAnalytics/.build/checkouts/firebase-ios-sdk/SwiftPM-PlatformExclude/FirebaseFunctionsWrap/include' not found
    umbrella "/Users/overcot/Work/kinbery-ios/SAF/SAFAnalytics/.build/checkouts/firebase-ios-sdk/SwiftPM-PlatformExclude/FirebaseFunctionsWrap/include"
             ^
If you have a downloadable sample project that reproduces the bug you're reporting, you will
likely receive a faster response on your issue.

#### Relevant Code:

```
// TODO(you): code here to reproduce the problem
```

**Comment by ryanwilson** [comment]
Thanks for the report, and sorry for the troubles.  In this case it's likely that there's some cached build artifacts relying on an older version of Firebase - the `FirebaseFunctionsWrap` directory doesn't exist anymore (was removed in #8864). 

Can you try to clean any cached build artifacts, make sure you're up to date, and try again? Thanks!

**Comment by Overcot** [comment]
Thanks for quick reply. 
Yeah, it was cache inside .build folder inside my package
But I receive another warnings after invocating
```swift
swift build
```

Are they intended?

[21961/21961] Downloading binary artifacts
/Users/overcot/Work/kinbery-ios/SAF/SAFAnalytics/.build/checkouts/firebase-ios-sdk/SwiftPM-PlatformExclude/FirebaseDynamicLinksWrap/dummy.m:17:2: warning: "Firebase Dynamic Links only supports the iOS platform" [-W#warnings]

 ^
1 warning generated.
/Users/overcot/Work/kinbery-ios/SAF/SAFAnalytics/.build/checkouts/firebase-ios-sdk/SwiftPM-PlatformExclude/FirebaseAnalyticsWrap/dummy.m:17:2: warning: "Firebase Analytics only supports the iOS platform" [-W#warnings]
#warning "Firebase Analytics only supports the iOS platform"
 ^
1 warning generated.

**Comment by paulb777** [comment]
The Dynamic Links warning is correct. The Analytics warning should have been changed to match to only trigger for watchOS like this one https://github.com/firebase/firebase-ios-sdk/blob/master/SwiftPM-PlatformExclude/FirebaseAnalyticsSwiftWrap/dummy.m#L16. We'll do a fix. For now, it's safe to ignore.

Note that because `swift build` only supports macOS, we generally do builds with `xcodebuild`.

**Comment by paulb777** [comment]
Fixed in #9029. Thanks for the report @Overcot !

**Comment by Overcot** [comment]
Thanks for quick fix @paulb777! LFW to see it in incoming releases :)
